dali/pipeline/operators/decoder/nvjpeg_decoder_new.cu

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/pipeline/operators/decoder/nvjpeg_decoder_new.h"
+
+namespace dali {
+
+DALI_REGISTER_OPERATOR(nvJPEGDecoderNew, nvJPEGDecoderNew, Mixed);
```

Comment:
I believe we wan't to keep the name `nvJPEGDecoder` in both old and new version, and choose this one or the other at compile time (during CMake stage)

Reply:
Yes and no. The old one definitely consumes less memory than this one where for every image we create a pinned buffer.
Ideally, I think we should have:
* old one for the compatibility
* old one with the new API (and just num of threads set of buffers)
* new one, with the fully separated stages

This PR is something in between 2 and 3.

Reply:
> The old one definitely consumes less memory than this one where for every image we create a pinned buffer.
There is no need to keep pinned buffer for every image using new API - it can be done similarly to old API. I'll follow up with Serge on that

Reply:
nvJPEGDecoderNew is very ambiguous for the user. Should we have more meaningful names?

Reply:
@marsaev I'll change to have one pinned buffer per thread